pulsar-client-cpp/include/pulsar/MessageId.h

```diff
@@ -28,6 +28,7 @@
 namespace pulsar {
 
 class MessageIdImpl;
+typedef std::shared_ptr<MessageIdImpl> MessageIdImplPtr;
```

Comment:
Is this `typedef` used anywhere?

Reply:
Cover 5 files and use 17 times

Reply:
I mean the headers under `include/pulsar/` are exposed to users. It's not a good manner to introduce a new `typedef` in these headers. You can add the common `typedef`s to `lib/xxx.h`, which is not exposed to users.

BTW, I see the duplicated `typedef` in `ChunkMessageIdImpl.h`.

Reply:
I put typedef in MessageId.h